lib/internal/modules/esm/loader.js

```diff
@@ -328,36 +306,116 @@ class ESMLoader {
+  }
+
+  /**
+   * Resolve the location of the module.
+   *
+   * @param {string} originalSpecifier The specified URL path of the module to
+   *                                   be resolved.
+   * @param {string} [parentURL] The URL path of the module's parent.
+   * @param {ImportAssertions} importAssertions Assertions from the import
+   *                                            statement or expression.
+   * @returns {{ format: string, url: URL['href'] }}
+   */
+  resolve(originalSpecifier, parentURL, importAssertions) {
+    return hooksProxy.makeSyncRequest('resolve', originalSpecifier, parentURL, importAssertions);
+  }
+
+  async #getModuleJob(specifier, parentURL, importAssertions) {
+    const resolveResult = await hooksProxy.makeAsyncRequest('resolve', specifier, parentURL, importAssertions);
+    return this.getJobFromResolveResult(resolveResult, parentURL, importAssertions);
+  }
+  getModuleJob(specifier, parentURL, importAssertions) {
+    const jobPromise = this.#getModuleJob(specifier, parentURL, importAssertions);
+    return {
+      run() {
+        return PromisePrototypeThen(jobPromise, (job) => job.run());
+      },
+      get modulePromise() {
+        return PromisePrototypeThen(jobPromise, (job) => job.modulePromise);
+      },
+      get linked() {
+        return PromisePrototypeThen(jobPromise, (job) => job.linked);
+      },
+    };
+  }
+
+  /**
+   * Provide source that is understood by one of Node's translators.
+   *
+   * @param {URL['href']} url The URL/path of the module to be loaded
+   * @param {object} [context] Metadata about the module
+   * @returns {Promise<{ format: ModuleFormat, source: ModuleSource }>}
+   */
+  async load(url, context) {
+    const result = await hooksProxy.makeAsyncRequest('load', url, context);
+    this.validateLoadResult(url, result?.format);
+    return result;
   }
 }
 
-ObjectSetPrototypeOf(ESMLoader.prototype, null);
 
-exports.ESMLoader = ESMLoader;
+let emittedExperimentalWarning = false;
+/**
+ * A loader instance is used as the main entry point for loading ES modules. Currently, this is a singleton; there is
+ * only one used for loading the main module and everything in its dependency graph, though separate instances of this
+ * class might be instantiated as part of bootstrap for other purposes.
+ * @param {boolean} useCustomLoadersIfPresent If the user has provided loaders via the --loader flag, use them.
+ * @returns {instanceof BaseModuleLoader | instanceof CustomizedModuleLoader}
+ */
+function createModuleLoader(useCustomLoadersIfPresent = true) {
+  if (useCustomLoadersIfPresent) {
+    const userLoaderPaths = getOptionValue('--experimental-loader');
+    if (userLoaderPaths.length > 0) {
+      if (!emittedExperimentalWarning) {
+        emitExperimentalWarning('Custom ESM Loaders');
+        emittedExperimentalWarning = true;
+      }
+      return new CustomizedModuleLoader(userLoaderPaths);
```

Comment:
The `userLoadersPaths` variable does not exist in the constructor signature of `CustomizedModuleLoader` and `DefaultModuleLoader` classes. Is it right or I'm missing something?